src/libstapsdt.h

```diff
@@ -47,10 +47,20 @@ typedef struct SDTProvider {
   // private
   void *_handle;
   char *_filename;
+  int _memfd;
+  int _use_memfd;
 } SDTProvider_t;
 
 SDTProvider_t *providerInit(const char *name);
 
+/*
+Linux newer than 3.17 with libc that supports the syscall it will default to
+using a memory-backed file descriptor. This behavior can be overridden at
+runtime by calling this with use_memfd = 0 prior after providerInit, and before
+providerLoad.
+*/
+int providerUseMemfd(SDTProvider_t *provider, const int use_memfd);
```

Comment:
because it is pure C, we can't overload the providerInit function like I'd normally prefer to, so I added a helper instead.